pandas/io/sql.py

```diff
@@ -2069,6 +2080,18 @@ def drop_table(self, table_name: str, schema: str | None = None) -> None:
                 self.get_table(table_name, schema).drop(bind=self.con)
             self.meta.clear()
 
+    def delete_rows(self, table_name: str, schema: str | None = None) -> None:
+        schema = schema or self.meta.schema
+        if self.has_table(table_name, schema):
+            self.meta.reflect(
+                bind=self.con, only=[table_name], schema=schema, views=True
+            )
+            with self.run_transaction():
```

Comment:
Why do we need to run a transaction here? Can this be done without changing anything at the COMMIT level?

Reply:
Hello :)
I did some investigation and found this pattern `with self.run_transaction()` dates back to this [PR](https://github.com/pandas-dev/pandas/pull/48576). Maybe @mroeschke can add some color in here ?

I'm probably wrong but couldn't find a reason to keep it. Module's `to_sql` always creates a `pandasSQL_builder` with `need_transaction=True` meaning everything running within the class should already be in a transaction as per these two [lines](https://github.com/pandas-dev/pandas/blob/19ea997815d4dadf490d7052a0a3c289be898588/pandas/io/sql.py#L1634-L1635) ?
**TLDR**: do we need `run_transaction` under SQLAlchemy > 2 ?

Reply:
By the way, it has worked without a transaction block 👌

Reply:
Sounds good - let's remove the block then. Generally we don't want to add transactions on behalf of the user

Reply:
Alright ! I'll wait for CI to complete and open the PR in case it works :). Thanks a ton